client/src/app/components/Pagination.tsx

```diff
@@ -0,0 +1,170 @@
+import classNames from "classnames";
+import React, { Component, ReactNode } from "react";
+import "./Pagination.scss";
+import { PaginationProps } from "./PaginationProps";
+import { PaginationState } from "./PaginationState";
+
+/**
+ * Component which will display pagination.
+ */
+class Pagination extends Component<PaginationProps, PaginationState> {
+    /**
+     * Dots for pagination.
+     */
+     private static readonly DOTS: string = "...";
+
+    /**
+     * Create a new instance of Pagination.
+     * @param props The props.
+     */
+    constructor(props: PaginationProps) {
+        super(props);
+        this.state = {
+            paginationRange: [],
+            lastPage: 0
+        };
+    }
+
+    /**
+     * The component updated.
+     * @param prevProps previous props
+     */
+    public componentDidUpdate(prevProps: PaginationProps): void {
+        if (this.props !== prevProps) {
```

Comment:
Shouldn't we be checking just on the specific property? Props could have a lot of other properties that we are not interested in watching here.